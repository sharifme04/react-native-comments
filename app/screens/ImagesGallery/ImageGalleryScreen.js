import React from 'react';
import {SafeAreaView, ScrollView, StyleSheet, Text, View} from 'react-native';
import ImageElement from '../../components/ImageElement';
import styles from './ImageGalleryStyle';

const ImageGalleryScreen = ({images, handleVisible, handleImage}) => {
  //console.log('ImageGalleryScreen  11', images);
  const imagesList = images?.map(e => (
    <ImageElement
      key={e.id}
      imageUrl={e?.src?.large2x}
      id={e.id}
      handleVisible={handleVisible}
      handleImage={handleImage}
    />
  ));

  return (
    <View>
      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={{
          flexDirection: 'row',
          flexWrap: 'wrap',
        }}>
        {imagesList}
      </ScrollView>
    </View>
  );
};

export default ImageGalleryScreen;
