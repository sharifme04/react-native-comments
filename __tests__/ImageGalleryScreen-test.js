import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImageGalleryScreen from '../app/screens/ImagesGallery/ImageGalleryScreen';

const fakerimages = [
  {
    alt: 'Person Walking Between Green Forest Trees',
    avg_color: '#283419',
    height: 1667,
    id: 15286,
    liked: false,
    photographer: 'Luis del RÃ­o',
    photographer_id: 1081,
    photographer_url: 'https://www.pexels.com/@luisdelrio',
    src: {},
    url: 'https://www.pexels.com/photo/person-walking-between-green-forest-trees-15286/',
    width: 2500,
  },
  {
    alt: 'Scenic View Of Snow Capped Mountains During Night',
    avg_color: '#557088',
    height: 2255,
    id: 3408744,
    liked: false,
    photographer: 'stein egil liland',
    photographer_id: 144244,
    photographer_url: 'https://www.pexels.com/@therato',
    src: {},
    url: 'https://www.pexels.com/photo/scenic-view-of-snow-capped-mountains-during-night-3408744/',
    width: 3546,
  },
];

test('ImageGalleryScreen ', () => {
  const tree = renderer.create(<ImageGalleryScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ImageGalleryScreen empty images ', () => {
  const tree = renderer.create(<ImageGalleryScreen images={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ImageGalleryScreen with images ', () => {
  const tree = renderer
    .create(
      <ImageGalleryScreen
        handleVisible={() => {}}
        handleImage={() => {}}
        images={fakerimages}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
