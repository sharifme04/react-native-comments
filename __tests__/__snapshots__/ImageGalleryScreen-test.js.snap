// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGalleryScreen  1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    style={
      Object {
        "backgroundColor": "#eee",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ImageGalleryScreen empty images  1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    style={
      Object {
        "backgroundColor": "#eee",
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`ImageGalleryScreen with images  1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
    style={
      Object {
        "backgroundColor": "#eee",
      }
    }
  >
    <View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 325.5,
            "margin": 2,
            "padding": 2,
            "width": 371,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "width": null,
            }
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 325.5,
            "margin": 2,
            "padding": 2,
            "width": 371,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": undefined,
            }
          }
          style={
            Object {
              "alignSelf": "stretch",
              "flex": 1,
              "width": null,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
